dace/transformation/passes/constant_propagation.py

```diff
@@ -69,40 +74,85 @@ def apply_pass(self, sdfg: SDFG, _, initial_symbols: Optional[Dict[str, Any]] =
         if not initial_symbols and not self.should_apply(sdfg):
             result = {}
         else:
-            # Trace all constants and symbols through states
-            per_state_constants: Dict[SDFGState, Dict[str, Any]] = self.collect_constants(sdfg, initial_symbols)
+            arrays: Set[str] = set(sdfg.arrays.keys() | sdfg.constants_prop.keys())
+
+            # Add nested data to arrays
+            def _add_nested_datanames(name: str, desc: data.Structure):
+                for k, v in desc.members.items():
+                    if isinstance(v, data.Structure):
+                        _add_nested_datanames(f'{name}.{k}', v)
+                    elif isinstance(v, data.ContainerArray):
+                        # TODO: How are we handling this?
```

Comment:
I know this was moved from another function, but:
Should be fine as-is, no new names are added. Unless you want to add datanames from `v.stype` if the internal array has a `Structure`... Unrelated to this PR, I don't think the `a.b` array names really work.